driver-core/src/main/com/mongodb/connection/ByteBufBsonDocument.java

```diff
@@ -58,7 +58,17 @@
     }
 
     static ByteBufBsonDocument createOne(final ByteBufferBsonOutput bsonOutput, final int startPosition) {
-        return create(bsonOutput, startPosition).get(0);
+        List<ByteBuf> duplicateByteBuffers = bsonOutput.getByteBuffers();
```

Comment:
This change is necessary because otherwise calling createOne on a document within an OP_MSG that is followed by a payload will fail, as the create method thinks it's just a bunch of contiguous documents and doesn't know anything about the payload structure.  So this is just a copy of createOne but without the loop.  

I didn't see a straightforward way to refactor for code re-use, but I'll take another look...

Reply:
Seems simple enough.